File: CSharp/_VIPER Manutenção_/Code/Routers/VIPERRouter.cs
//
//  __MODULENAME__Router.cs
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

using NAMESPACE.Interactors;
using NAMESPACE.Interfaces;
using NAMESPACE.Presenters;
using NAMESPACE.Views;
using System.Windows.Forms;

namespace NAMESPACE.Routers
{
    public class VIPERRouter : IPresenterToRouterVIPER
    {
        public static Form New()
        {
            return new VIPERRouter().Inicializar();
        }

        public Form Inicializar()
        {
            VIPERPresenter presenter = new VIPERPresenter();
            VIPERInteractor interactor = new VIPERInteractor();
            VIPERRouter router = new VIPERRouter();

            VIPERView form = new VIPERView();
            form.presenter = presenter;

            presenter.interactor = interactor;
            presenter.router = router;
            presenter.view = form;
            interactor.presenter = presenter;

            return form;
        }
		
		public Form LoadModule()
        {
            throw new System.NotImplementedException();
        }
    }
}

File: Delphi/$Padrão Manutenção Com Código Fonte$/Codigo/routers/VIPERRouter.pas
{
  __MODULENAME__Router.pas
  __APPNAME__

  Created by __USERNAME__ on __DATETIME__.
  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
}

unit VIPERRouter;

interface

uses
  VIPERInterfaces, VIPERView, Forms;

type
  TVIPERRouter = class(TInterfacedObject, IPresenterToRouterVIPER)
    class function New: IPresenterToRouterVIPER;
    procedure LoadModule(View: TForm);
  end;

implementation

{ TVIPERRouter }

uses VIPERPresenter, VIPERInteractor;

procedure TVIPERRouter.LoadModule(View: TForm);
begin
  var Presenter := TVIPERPresenter.Create;
  var Interactor := TVIPERInteractor.Create;
  var Router := TVIPERRouter.Create;

  TFrmVIPERView(View).Presenter := Presenter;
  Presenter.View := TFrmVIPERView(View);
  Presenter.Router := Router;
  Presenter.Interactor := Interactor;
  Interactor.Presenter := Presenter;
end;

class function TVIPERRouter.New: IPresenterToRouterVIPER;
begin
  Result := TVIPERRouter.Create;
end;

end.
